### PR TITLE
B 20010 int Add FF for HQ role 

### PR DESCRIPTION
**Pull request by r-mettler**
## Summary

The changes here are to FF off the Headquarters Role in the Admin App Office Account creation page and remove the HQ role checkbox from the Office App Request Account page. Since FF can't currently be read without being signed in the Office App will get the HQ role checkbox back once we determine all HQ role options are functioning properly. For now the only way to create a HQ user role is via the Admin App Office Creation page with the FEATURE_FLAG_HEADQUARTERS_ROLE feature flag turned on.

This backlog adds a new Headquarters Office user role. Further downstream tasks will add privileges meant for the HQ role which will limit pages to read only.

Is there anything you would like reviewers to give additional scrutiny?

Turn the FEATURE_FLAG_HEADQUARTERS_ROLE on/off and test. 

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Login to the admin application
2. Create a new Office user
3. Select the Safety Moves Privilege then attempt to select HQ checkbox you should NOT be able to
4. Deselect Safety Moves Privilege then select HQ role
5. Attempt to select Safety Moves Privilege you should NOT be able to
6. Select the TOO role and click Save

image (2)

7. Go to the Office app and sign in as the newly created user
8. Click the change user role and verify you can login with the HQ role
image (3)
NOTE: At this point you should be able to do what a TOO can do but this story was to just allow a new login role

9. **Turn off** the **FEATURE_FLAG_HEADQUARTERS_ROLE** by changing it to false in the **.envrc** file
10. Also turn off **headquarters_role** in the **development.features.yaml** file by setting **enabled** and **value** to **false** for the headquarters_role key.
11. Kill your server and client in the terminal
12. Run direnv allow
13. Relaunch server and client in terminal
14. Login to the Admin app and go to create a new Office User
15. Fill out the form then attempt to select the Headquarters role checkbox you should NOT be able to

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

### CODE REVIEW

File: src/pages/Office/RequestAccount/RequestAccount.jsx [hunk 1]
```diff
@@ -62,10 +62,10 @@ export const RequestAccount = ({ setFlashMessage }) => {
         roleType: 'qae_csr',
       });
     }
-    if (values.headquartersCheckBox) {
+    if (values.customerSupportRepresentativeCheckBox) {
```
Comment (ajlusk) [inline src/pages/Office/RequestAccount/RequestAccount.jsx:65]: Looks like a weird merge thing caused this `if` block to be duplicated.

**Review by ajlusk — APPROVED** [review]
Functions as described. Tests are passing.

Just needs that duplicate if statement removed but that shouldn't affect functionality.

File: src/pages/Office/RequestAccount/RequestAccount.jsx (hunk 1 above)
Comment (r-mettler) [reply src/pages/Office/RequestAccount/RequestAccount.jsx:65]: Ha, figures. It didn't even flag it as a conflict. I'll go fix that.

Comment (r-mettler) [reply src/pages/Office/RequestAccount/RequestAccount.jsx:65]: Fixed it. Only did that on this INT branch. Thanks git *sigh

**Comment by ajlusk** [comment]
Undoing approval.

When I create a new HQ user from [http://officelocal:3000/devlocal-auth/login](http://officelocal:3000/devlocal-auth/login) with the FF enabled it only has the TOO role.

**Review by ajlusk — APPROVED** [review]
Never mind, with the comment above. No need to hold up this any longer. I'm adding a fix for it in my PR, it could reasonably be part of either.

File: src/scenes/SystemAdmin/shared/RolesPrivilegesCheckboxes.jsx [hunk 2]
```diff
@@ -40,6 +51,22 @@ const RolesPrivilegesCheckboxInput = (props) => {
       }
     }
 
+    if (!isHeadquartersRoleFF && input.includes('headquarters')) {
```
Comment (MInthavongsay) [inline src/scenes/SystemAdmin/shared/RolesPrivilegesCheckboxes.jsx:54]: Is there a constant for these potential values?  There are multiple lines we are hardcoding 'headquarters'.

**Review by MInthavongsay — APPROVED** [review]
Looks good. Just one minor CONSTANTS usage comment.

File: src/scenes/SystemAdmin/shared/RolesPrivilegesCheckboxes.jsx (hunk 2 above)
Comment (r-mettler) [reply src/scenes/SystemAdmin/shared/RolesPrivilegesCheckboxes.jsx]: There is a roleTypes constant that is used to setup all these roles and those values match what is here so I made the change to check against constants.

**Pull request merged**